packages/docs/build/bots/interactions.mdx

```diff
@@ -0,0 +1,787 @@
+    for (const component of form.components) {
+      if (component.component.case === 'button') {
+        if (component.id === 'yes') poll.yes++
+        if (component.id === 'no') poll.no++
+      }
+    }
+    
+    await handler.sendMessage(
+      event.channelId,
+      `Results: Yes: ${poll.yes}, No: ${poll.no}`
+    )
+  }
+})
+```
+
+### Multi-Step Form
+
+```ts
+const userState = new Map<string, { step: string; data: any }>()
+
+bot.onSlashCommand('register', async (handler, event) => {
+  userState.set(event.userId, { step: 'name', data: {} })
+  
+  await handler.sendInteractionRequest(
+    event.channelId,
+    {
+      case: 'form',
+      value: {
+        id: 'register-step1',
+        title: 'Registration (Step 1/2)',
+        subtitle: 'Enter your name:',
+        components: [
+          {
+            id: 'name',
+            component: {
+              case: 'textInput',
+              value: { label: 'Full Name', placeholder: 'John Doe' }
+            }
+          },
+          { id: 'next', component: { case: 'button', value: { label: 'Next' } } }
+        ]
+      }
+    },
+    hexToBytes(event.userId as `0x${string}`)
+  )
+})
+
+bot.onInteractionResponse(async (handler, event) => {
+  if (event.response.payload.content?.case !== 'form') return
+  
+  const form = event.response.payload.content.value
+  const state = userState.get(event.userId)
+  
+  if (!state) return
+  
+  if (form.id === 'register-step1' && state.step === 'name') {
+    // Get name input
+    for (const component of form.components) {
+      if (component.component.case === 'textInput' && component.id === 'name') {
+        state.data.name = component.component.value.value
+      }
+    }
+    
+    // Move to step 2
+    state.step = 'email'
+    
+    await handler.sendInteractionRequest(
+      event.channelId,
+      {
+        case: 'form',
+        value: {
+          id: 'register-step2',
+          title: 'Registration (Step 2/2)',
+          subtitle: 'Enter your email:',
+          components: [
+            {
+              id: 'email',
+              component: {
+                case: 'textInput',
+                value: { label: 'Email', placeholder: 'you@example.com' }
+              }
+            },
+            { id: 'submit', component: { case: 'button', value: { label: 'Submit' } } }
+          ]
+        }
+      },
+      hexToBytes(event.userId as `0x${string}`)
+    )
+  } else if (form.id === 'register-step2' && state.step === 'email') {
+    // Get email and complete registration
+    for (const component of form.components) {
+      if (component.component.case === 'textInput' && component.id === 'email') {
+        state.data.email = component.component.value.value
+      }
+    }
+    
+    await handler.sendMessage(
+      event.channelId,
+      `Registration complete! Name: ${state.data.name}, Email: ${state.data.email}`
+    )
+    
+    userState.delete(event.userId)
+  }
+})
+```
+
+## Best Practices
```

Comment:
Can this be a summary list instead of writing good vs bad examples? We can make it shorten 
```
- Use button labels to be descriptive about the action
- Provide context with subtitles
- Use targeted and public interactions accordingly 
```